packages/generator/src/generator.ts

```diff
@@ -55,33 +54,47 @@ const logger = createLogger({
   messageContext: 'generator'
 });
 
-export async function generate(
-  options: GeneratorOptions
-): Promise<void | EmitResult[]> {
+const threadsForCompiling = 16;
+
+export async function generate(options: GeneratorOptions): Promise<void> {
   const project = await generateProject(options);
   if (!project) {
     throw Error('The project is undefined.');
   }
 
   await project.save();
-
   if (options.generateJs) {
-    project
+    let directories = project
       .getDirectories()
-      .filter(dir => !!dir.getSourceFile('tsconfig.json'))
-      .forEach(dir => {
-        logger.info(`Transpiling files in the directory: ${dir.getPath()}.`);
-
-        execa('tsc', { cwd: dir.getPath() }).catch(err => {
-          logger.error(`Error: Failed to generate js files: ${err}`);
-          process.exit(1);
-        });
-      });
+      .filter(dir => !!dir.getSourceFile('tsconfig.json'));
+    while(directories.length > 0){
+      const chunk = directories.splice(0,threadsForCompiling)
```

Comment:
I am not a big fan of changing the input here(`directories`) with splice. Especially with my comment above to put into a separate function it will have a side effect. Instead consider something like this:
```js
const chunk = (d) => {
  let chunks = [];
  for(let i = 0; i < d.length/threadsForCompiling; i++) {
    chunks = [...chunks, d.slice(i * threadsForCompiling, (i + 1) * threadsForCompiling)];
  }
  return chunks;
}
```

Reply:
I had the same initial hesitation, but thought then: this is so easy and feels so 2012 Physics code doing things in parallel using MPI or openMP. But you are right and we should have pure functions. No side effects. So I make at least a shallow copy with slice.

Reply:
I also agree that it is a little easier to read. Maybe just do it in a function and then:
`function (input) { const directoriesYouWorkWith = [...input] }` and then keep it as is?

Reply:
I created a splitInChunks in the array util. Then this can be used somewhere else perhaps. I think it is nicer now.